src/Microsoft.AspNet.OData.Shared/Query/Expressions/FilterBinder.cs

```diff
@@ -335,8 +335,15 @@ private Expression BindSingleResourceCastFunctionCall(SingleResourceFunctionCall
                 // We only support to cast Entity type to the same type now.
                 return arguments[0];
             }
-            else
+            else if (targetClrType.IsSubclassOf(arguments[0].Type))
```

Comment:
>IsSubclassOf [](start = 35, length = 12)

I think we generally use IsAssignableFrom.  The key difference is that IsAssignableFrom works with interfaces, where IsSubclassOf does not. I don't know that we currently support interfaces all through the stack, but I'd like to be able to in the future.

Reply:
Yes you are right, would be better to use IsAssignableForm thinking futuristic once we support interfaces, #Resolved